### ISSUE TITLE
A rebooting occured when running "06-mutex/esp32-freertos-06-solution-task-parameters"

### ISSUE THREAD

**Issue by wanggaoteng** [ShawnHymel/introduction-to-rtos#10]
Hi, there is a rebooting when running the "06-mutex/esp32-freertos-06-solution-task-parameters" in esp32-s3:

`assert failed: xQueueGenericSend queue.c:832 (pxQueue->pcHead != ((void *)0) || pxQueue->u.xSemaphore.xMutexHolder == ((void *)0) || pxQueue->u.xSemaphore.xMutexHolder == xTaskGetCurrentTaskHandle())
`

![image](https://github.com/user-attachments/assets/510d0059-7d70-480e-b17d-b13122b863fd)

Is there anyway to fix it?

Best regards.

**Comment by wanggaoteng** [comment]
After many attempts, I find there is a [pull request](https://github.com/ShawnHymel/introduction-to-rtos/pull/9) from [morgoob](https://github.com/morgoob), by referring to the codes, the problem solved, and test in esp32-s3. 
Thanks @morgoob

`
#include <Arduino.h>
// Use only core 1 for demo purposes
#if CONFIG_FREERTOS_UNICORE
  static const BaseType_t app_cpu = 0;
#else
  static const BaseType_t app_cpu = 1;
#endif

// Pins (change this if your Arduino board does not have LED_BUILTIN defined)
static const int led_pin = 21;

// Globals
static SemaphoreHandle_t mutex;

//*****************************************************************************
// Tasks

// Blink LED based on rate passed by parameter
void blinkLED(void *parameters) 
{
  // Copy the parameter into a local variable
  int num = *(int *)parameters;

  // Release the mutex so that the creating function can finish
  // Release the mutex so that setup() function can finish
  xSemaphoreGive(mutex);

  // Print the parameter
  Serial.print("Received: ");
  Serial.println(num);

  // Configure the LED pin
  pinMode(led_pin, OUTPUT);

  // Blink forever and ever
  while (1) 
  {
    digitalWrite(led_pin, HIGH);
    vTaskDelay(num / portTICK_PERIOD_MS);
    digitalWrite(led_pin, LOW);
    vTaskDelay(num / portTICK_PERIOD_MS);
  }
}

//*****************************************************************************
// Main (runs as its own task with priority 1 on core 1)

void setup() 
{
  long int delay_arg;

  // Configure the LED pin
  pinMode(led_pin, OUTPUT);

  // Configure Serial
  Serial.begin(9600);

  // Wait a moment to start (so we don't miss Serial output)
  vTaskDelay(1000 / portTICK_PERIOD_MS);
  Serial.println();
  Serial.println("---FreeRTOS Mutex Solution---");
  Serial.println("Enter a number for delay (milliseconds)");
  // Wait for input from Serial
  while (Serial.available() <= 0);
  // Read integer value
  delay_arg = Serial.parseInt();
  Serial.print("Sending: ");
  Serial.println(delay_arg);
  
  // Create mutex before starting tasks
  mutex = xSemaphoreCreateMutex();

  // Start task 1
  xTaskCreatePinnedToCore(blinkLED,
                          "Blink LED",
                          1024,
                          (void *)&delay_arg,
                          1,
                          NULL,
                          app_cpu);

  // Do nothing until mutex has been returned (maximum delay)
  // one tick delay to make sure mutex is acquired by other task
  vTaskDelay(portTICK_PERIOD_MS);

  // take mutex (this will be blocked until blinkLED releases it)
  xSemaphoreTake(mutex, portMAX_DELAY);

  // Show that we accomplished our task of passing the stack-based argument
  Serial.println("Done!");
}
void loop() 
{
  // Do nothing but allow yielding to lower-priority tasks
  vTaskDelay(1000 / portTICK_PERIOD_MS);
}
`
